docs/adoc/config-product.adoc

```diff
@@ -0,0 +1,325 @@
+
+Each `<language/>` element defines `<deliverable/>` elements, that is, output documents.
+Choose which formats to enable for which document using the `<format/>`.
+
+Subsequent `<language/>` elements, that is, references to translations can use two modes:
+* `@translation-type="full"`: All documents available in the default language will also be generated for the translation.
+
+* `@translation-type="list"`: Use `<deliverable/>` elements to select those documents from the default language which have a translation.
+Within a `<deliverable/>` element of a non-default `<language/>` element, the number of elements available is restricted to `<dc/>` and `<subdeliverable/>`, other parameters will be
+
+
+[#config-product-external-reference]
+### References to documents not generated within {ds2}, `<external/>`
+
+The `<external/>` element collects links to documents that are not generated using {ds2}.
+{ds2} makes no effort to ensure availability of such documents, meaning all links to them must be double-checked.
+Such documents can be co-hosted along with {ds2} or be on another host altogether.
+
+For each required document, add a `<link/>` element to the `<external/>` element.
+A `<link/>` can reference documents in multiple formats and multiple languages.
+The primary language (`<language default="true"/>`) must be listed first.
+
+
+[#config-product-internal-reference]
+### {ds2}-internal references, `<internal/>`
+
+Within the `<internal/>` element, reference documents already defined elsewhere in the {ds2} configuration
+Document references (using the `<ref/>` element) will inherit all properties of the original document, such as available formats and languages and reduce duplication.
+You can also reference related product pages within this element.
+
+References identify the document using the product ID, docset ID, and DC file/link ID.
+
+A `<ref/>` can have the following structures:
+
+* A reference to the default docset of a product: `<ref product="__PRODUCT_ID__"/>`
+
+* A reference to a specific docset of a product: `<ref product="__PRODUCT_ID__" docset="__SET_ID__"/>`
+
+* A reference to a specific DC definition within `<builddocs/>`: `<ref product="__PRODUCT_ID__" docset="__SET_ID__" dc="__DC_NAME__"/>`
+
+* A reference to a specific DC definition that has subdeliverables within `<builddocs/>`: `<ref product="__PRODUCT_ID__" docset="__SET_ID__" dc="__DC_NAME__" subdeliverable="__SUBDELIVERABLE_NAME__"/>`
+
+* A reference to a specific external link: `<ref product="__PRODUCT_ID__" docset="__SET_ID__" link="__LINK_ID__"/>`
+The value for `__LINK_ID__` has to be specifically set on the corresponding `<link/>` element by adding the attribute `<link linkid="__NAME_OF_ID__"/>`. `__NAME_OF_ID__ must be unique within the current `<docset/>` and may consist of talphanumeric characters and the characters `_-.`.
```

Comment:
Typo.

```suggestion
The value for `__LINK_ID__` has to be specifically set on the corresponding `<link/>` element by adding the attribute `<link linkid="__NAME_OF_ID__"/>`. `__NAME_OF_ID__ must be unique within the current `<docset/>` and may consist of alphanumeric characters and the characters `_-.`.
```

Reply:
Thanks! I resolved this locally after you told me via chat before I saw this comment. Resolving.